proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/verifyserviceprovider/TranslatedHubResponse.java

```diff
@@ -49,7 +54,7 @@ public VspLevelOfAssurance getLevelOfAssurance() {
         return levelOfAssurance;
     }
 
-    public Attributes getAttributes() {
-        return attributes;
+    public Optional<Attributes> getAttributes() {
+        return fromNonMatchingAttributes(attributes);
```

Comment:
Would `attributes` ever be null though since we have the `@NotNull` flag above?

Reply:
🤔 hmm, that's a good point. The main reason I'm handling the null case is due to [this test](https://github.com/alphagov/verify-proxy-node/blob/eid-1762-consolidate-user-attribute-hash-logging-logic/proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/VspProxyTranslateResponseTest.java#L101) which suggests that if authentication failed, you'd get a response with the attributes as null...

Reply:
I've checked and VSP will return `null` for attributes, as well as `pid`, and  `loa`. I'm not sure what good the @NotNull flag is doing. Does it even ever get validated?

Reply:
It doesn't. I don't think the annotations are actually doing anything. I've removed them.

Reply:
huh interesting. Good job for handling the null case 👍

Reply:
Should we also make `getLevelOfAssurance` and `getPid` Optional then?